tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
-        [InlineData(TestConstants.B2CAuthority, TestConstants.B2CAuthorityWithV2)]
-        [InlineData(TestConstants.B2CCustomDomainAuthority, TestConstants.B2CCustomDomainAuthorityWithV2)]
-        public void EnsureAuthorityIsV2_0(string initialAuthority, string expectedAuthority)
-        {
-            OpenIdConnectOptions options = new OpenIdConnectOptions
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigActions_B2cSpecificSetup(bool useServiceCollectionExtension)
+        {
+            _configureMsOptions = (options) => {
+                options.Instance = TestConstants.AadInstance;
+                options.TenantId = TestConstants.TenantIdAsGuid;
+                options.ClientId = TestConstants.ClientId;
+                options.SignUpSignInPolicyId = TestConstants.B2CSignUpSignInUserFlow;
+                options.Domain = TestConstants.Domain;
+            };
+
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+
+            await AddSignIn_TestB2cSpecificSetup(services).ConfigureAwait(false);
+        }
+
+        private async Task AddSignIn_TestB2cSpecificSetup(IServiceCollection services)
+        {
+            var provider = services.BuildServiceProvider();
+
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            // Assert B2C name claim type
+            Assert.Equal(ClaimConstants.Name, oidcOptions.TokenValidationParameters.NameClaimType);
+
+            var httpContext = HttpContextUtilities.CreateHttpContext();
+            var authScheme = new AuthenticationScheme(OpenIdConnectDefaults.AuthenticationScheme, OpenIdConnectDefaults.AuthenticationScheme, typeof(OpenIdConnectHandler));
+            var authProperties = new AuthenticationProperties();
+            authProperties.Items[OidcConstants.PolicyKey] = TestConstants.B2CResetPasswordUserFlow;
+            var redirectContext = new RedirectContext(httpContext, authScheme, oidcOptions, authProperties)
+            {
+                ProtocolMessage = new OpenIdConnectMessage() { IssuerAddress = $"IssuerAddress/{TestConstants.B2CSignUpSignInUserFlow}/" },
+            };
+
+            await oidcOptions.Events.RedirectToIdentityProvider(redirectContext).ConfigureAwait(false);
+
+            // Assert issuer is updated to non-default user flow
+            Assert.Contains(TestConstants.B2CResetPasswordUserFlow, redirectContext.ProtocolMessage.IssuerAddress);
+        }
```

Comment:
I wouldn't use password reset here, as that policy will move under sign-up/sign-in soon, so maybe use edit profile or something else.
This is similar to the tests, expect you're setting the issuer address and the claim name...can you combine TestCommon and this? And then we will be asserting more values w/b2c as well.

Reply:
Nice to know. The specific values don't matter in this case but I updated it to Edit Profile for readability.